src/matching_market.sol

```diff
@@ -421,18 +427,33 @@ contract MatchingMarket is MatchingEvents, ExpiringMarket, DSNote {
 
     // ---- Internal Functions ---- //
 
-    function _buys(uint id, uint amount)
+    function _buys(uint id, uint quantity)
         internal
         returns (bool)
     {
-        require(buyEnabled);
-
-        if (amount == offers[id].pay_amt && isOfferSorted(id)) {
-            //offers[id] must be removed from sorted list because all of it is bought
+        OfferInfo memory offer = offers[id];
+        if(quantity < offer.pay_amt){
+            buy(id,quantity);
+        }else{
```

Comment:
put some spaces around else

Reply:
Done. And while doing it I found a bug, and corrected it in next push.
Thanks man!